Source/NSConcreteMapTable.m

```diff
@@ -116,16 +116,20 @@ @interface	NSConcreteMapTable : NSMapTable
 	else\
 	  pointerFunctionsAssign(&M->cb.pf.v, (void**)addr, (x).obj);
 */
+#define IS_WEAK_KEY(M) \
```

Comment:
Shouldn't these definitions come before the usages?

Reply:
They're macros, so they need only to come before macros that instantiate them are, themselves, instantiated.

Reply:
True for the compiler but confusing for me. Our most important goal should be to make the code easy to understand. For this moving the macros would really help.